Source/Scene/PointCloud3DTileContent.js

```diff
@@ -823,58 +713,310 @@ define([
             pass : isTranslucent ? Pass.TRANSLUCENT : Pass.OPAQUE,
             owner : content
         });
     }
 
+    function getStyleableProperties(source, properties) {
+        // Get all the properties used by this style
+        var regex = /czm_tiles3d_style_(\w+)/g;
+        var matches = regex.exec(source);
+        while (matches !== null) {
+            var name = matches[1];
+            if (properties.indexOf(name) === -1) {
+                properties.push(name);
+            }
+            matches = regex.exec(source);
+        }
+    }
+
+    function getVertexAttribute(vertexArray, index) {
+        var numberOfAttributes = vertexArray.numberOfAttributes;
+        for (var i = 0; i < numberOfAttributes; ++i) {
+            var attribute = vertexArray.getAttribute(i);
+            if (attribute.index === index) {
+                return attribute;
+            }
+        }
+    }
+
+    function createShaders(content, frameState, style) {
+        var i;
+        var name;
+        var attribute;
+
+        var context = frameState.context;
+        var batchTable = content.batchTable;
+        var hasBatchTable = defined(batchTable);
+        var hasStyle = defined(style);
+        var isQuantized = content._isQuantized;
+        var isOctEncoded16P = content._isOctEncoded16P;
+        var isRGB565 = content._isRGB565;
+        var isTranslucent = content._isTranslucent;
+        var hasColors = content._hasColors;
+        var hasNormals = content._hasNormals;
+        var hasBatchIds = content._hasBatchIds;
+        var backFaceCulling = content._backFaceCulling;
+        var vertexArray = content._drawCommand.vertexArray;
+
+        var colorStyleFunction;
+        var showStyleFunction;
+        var styleTranslucent = isTranslucent;
+
+        if (hasBatchTable) {
+            // Styling is handled in the batch table
+            hasStyle = false;
+        }
+
+        if (hasStyle) {
+            var shaderState = {
+                translucent : false
+            };
+            colorStyleFunction = style.getColorShaderFunction('getColorFromStyle', 'czm_tiles3d_style_', shaderState);
+            showStyleFunction = style.getShowShaderFunction('getShowFromStyle', 'czm_tiles3d_style_', shaderState);
+            styleTranslucent = shaderState.translucent;
+        }
+
+        content._styleTranslucent = styleTranslucent;
+
+        var hasColorStyle = defined(colorStyleFunction);
+        var hasShowStyle = defined(showStyleFunction);
+
+        // Get the properties in use by the style
+        var styleableProperties = [];
+        if (hasColorStyle) {
+            getStyleableProperties(colorStyleFunction, styleableProperties);
+        }
+        if (hasShowStyle) {
+            getStyleableProperties(showStyleFunction, styleableProperties);
+        }
+
+        // Disable vertex attributes that aren't used in the style, enable attributes that are
```

Comment:
The vertex array contains all the semantics and per-point properties, but styles may not use all the per-point properties, so those attributes are disabled. It may not matter in practice to disable unused attributes but it doesn't hurt.